src/Symfony/Component/Mailer/Transport/RoundRobinTransport.php

```diff
@@ -94,6 +97,17 @@ protected function isTransportDead(TransportInterface $transport): bool
 
     private function moveCursor(int $cursor): int
     {
-        return ++$cursor >= \count($this->transports) ? 0 : $cursor;
+        $cursor = ++$cursor >= \count($this->transports) ? 0 : $cursor;
+        $this->shuffleTransportsOrder($cursor);
+
+        return $cursor;
+    }
+
+    private function shuffleTransportsOrder(int $cursor = null)
+    {
+        if (false === $this->shuffle || 0 !== $cursor) {
```

Comment:
this is still wrong. If you loop back to 0, this would shuffle transports again. You should not shuffle transports when moving the cursor at all.

Reply:
Idea was to shake order when we're starting iterate them from 0

Reply:
but getting `0` as the current cursor does not mean you **start** iterating, as the iteration makes a cycle.